modules/BuildTools.Tasks/RunDotNet.cs

```diff
@@ -16,9 +18,35 @@ public class RunDotNet
 #else
 #error This must be built either for an SDK or for BuildTools
 #endif
-        : RunBase
+        : ToolTask
     {
-        protected override string GetExecutable()
+        /// <inheritdoc />
+        protected override MessageImportance StandardErrorLoggingImportance => MessageImportance.High;
+
+        /// <inheritdoc />
+        protected override MessageImportance StandardOutputLoggingImportance => MessageImportance.High;
+
+        protected override string ToolName => "dotnet";
+
+        /// <summary>
+        /// A list of arguments to be passed to the executable. The task will escape them for spaces and quotes.
+        /// </summary>
+        public ITaskItem[] Arguments { get; set; }
+
+        protected override string GenerateFullPathToTool()
+        {
+            return GetExecutable();
+        }
+
+        protected override string GenerateCommandLineCommands()
+        {
+            var arguments = string.Empty;
+            arguments += string.Join(" ", Arguments.Select(i => i.ItemSpec));
```

Comment:
Use ArgumentEscaper.EscapeAndConcatenate. String.Join(" ") doesn't work for arguments containing spaces or quotes.